### PR TITLE
Clarify syntax in all filtering documentation

### LINKED ISSUE #2310: Review response filtering docs and clarify syntax
### Description
Clarify syntax as specifically as possible for all response filtering documentation. See https://sensu.slack.com/archives/CCHNSF7RD/p1585173793002100. As written, the syntax may be difficult to follow. Consider explaining that filtering will not recognize values that start with a number as an identifier.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [x] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/5.19/dashboard/filtering/
https://docs.sensu.io/sensu-go/5.19/api/overview/#response-filtering
https://docs.sensu.io/sensu-go/5.19/sensuctl/reference/#response-filtering

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Added syntax notes in API response, sensuctl, and dashboard filtering documents.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2310

### CODE REVIEW

File: content/sensu-go/5.19/api/overview.md [hunk 1]
```diff
@@ -331,6 +332,24 @@ Here's the list of available fields:
...
+The filter statement construction is slightly different for different [operators][13], but there are a few general syntax rules that apply to all filter statements.
+For example:
+
+{{< highlight text >}}
+'fieldSelector=silenced.expire_on_resolve == true'
+{{< /highlight >}}
+
+As shown in this example:
+
+- Place the entire filter statement inside single or double quotes.
```
Comment (cwjohnston) [inline content/sensu-go/5.19/api/overview.md]: Feels like we're leaving something unsaid here. How do I know which to use? Do we need to link to an explainer?

Comment (cwjohnston) [reply content/sensu-go/5.19/api/overview.md]: Does this guideline also apply to API response filters? https://github.com/sensu/sensu-docs/pull/2348/files#diff-e9e39dba48f2c8082a60837d36897a68R101

**Review by cwjohnston — CHANGES_REQUESTED** [review]

**Comment by hillaryfraley** [comment]
Sounds like there is a rationale for single vs. double quotes in API response filtering (and I need to add it to the doc). See https://sensu.slack.com/archives/C60EEQFH8/p1586383336309000 for context.

**Comment by hillaryfraley** [comment]
@cwjohnston I updated the dashboard and API response filtering docs with clarifications from Eng. I decided it was better to just be opinionated and say "use single quotes" and explain the exception for shell vars rather than describe escaping double quotes, especially since I'm not clear on whether there is a good reason (besides the shell vars) to use double quotes. Do you think this is the right call?

**Review by cwjohnston — APPROVED** [review]

**Pull request merged**